[Product Collection] Fix: HTML entity decoding for product names in Hand-Picked Products

imanish003 opened the pull request:
<!-- Please do not remove any information from this pull request. Instead, add N/A or leave blank if not applicable -->

## What

I've implemented three updates in this PR:

- **Resolved the HTML Entities Decoding Issue:**
    - For instance, where it should say “Hoodie – Black”, it incorrectly displayed `Hoodie &#8211; Black`. The screenshot below illustrates this issue.

![image](https://github.com/woocommerce/woocommerce-blocks/assets/16707866/52b28cc9-3618-4ac9-8ea6-14a9af78c200)

- **Label Change for Consistency:**
    - Updated the label from “Pick some products” to “Hand-picked Products” for alignment with the filters dropdown.
- **Simplified by Removing the Description:**
    - I omitted the description as it wasn't significantly beneficial.

Fixes #

## Why

These changes were proposed by @manospsyx  on Slack (p1700652774848299-slack-C05UMTA7W0P).

<!-- Describe the reason for your changes. This will help the reviewer and future readers get additional context -->

## Testing Instructions

<!-- Write these steps from the perspective of a "user" (merchant) familiar with WooCommerce. No need to spell out the steps for common setup scenarios (eg. "Create a product"), but do be specific about the thing being tested. Include screenshots demonstrating expectations where that will be helpful. -->

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

**Checking the Hyphen Representation**:

1. Add a hyphen `–` to the name of any product, like **`Hoodie - Black`**.
2. Insert a Product Collection block into a new post.
3. Apply the “Hand-picked Products” filter in Inspector Controls.
4. Search for the product with the hyphen and confirm it displays correctly, as shown below:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/16707866/ae7c96e0-99fa-495b-8b83-1d0d0fbf3fde)

5. Feel free to test other [HTML entities](https://developer.mozilla.org/en-US/docs/Glossary/Entity) too.

**Verifying Label Update**:

- Ensure the text reads “HAND-PICKED PRODUCTS” instead of “PICK SOME PRODUCTS”, as depicted in the image below:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/16707866/66637971-647b-4fa7-a591-a211052fb61e)

* [ ] Do not include in the Testing Notes <!-- Check this box if this PR can't be tested (ie: it makes changes to tests, coding standards, docblocks, etc.). -->
* [ ] Should be tested by the development team exclusively <!-- Check this box if this PR should be tested by the development team exclusively (ie: it doesn't include user-facing changes or it can't be tested without manually modifying the code). -->

## Screenshots or screencast

<!-- Any screenshots of UI changes will be helpful to include here. Leave blank if not applicable. -->

| Before | After |
| ------ | ----- |
|        |       |

## WooCommerce Visibility

<!-- Check this documentation link (../docs/blocks/feature-flags-and-experimental-interfaces.md) to see if the change is visible in WooCommerce core, part of the feature plugin, or experimental. -->
Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:

* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:

* [x] This PR has a UI change and has been cross-browser tested at different viewport sizes on both the frontend and in the editor.
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog
<!-- Provide a brief, descriptive summary of the changes in this PR. Include potential impacts on different parts of the product. Example: "Updated the checkout process to streamline the experience for users and reduce the number of steps." -->

> [Product Collection] Fix: HTML entity decoding for product names in Hand-Picked Products

imanish003 committed:
Add HTML entity decoding for product names in Hand-Picked Products control

In the Hand-Picked Products control within the product-collection inspector controls, a function for decoding HTML entities in product names has been added.

- A new utility function `decodeHTMLEntities` has been implemented. This function decodes HTML entities in a string, ensuring that special characters are correctly displayed in their human-readable form.
- The `transformTokenIntoProductName` function has been updated to utilize `decodeHTMLEntities`. Now, when a product name is fetched (either directly as a token or via a product ID), the HTML entities within the name are decoded.
- This enhancement ensures that product names containing characters like ampersands or other HTML entities are accurately displayed in the UI.

This change improves the readability and accuracy of product names within the Hand-Picked Products control, enhancing the user experience for store managers using WooCommerce Blocks.

imanish003 committed:
Update label and hide description

This commit updates the `HandPickedProductsControl` component. Specifically, the user-facing label for product selection has been changed from 'Pick some products' to 'Hand-picked Products'. Additionally, the `__experimentalShowHowTo` property has been added with a `false` value, to hide description. Corresponding changes have been made in the E2E test file `product-collection.block_theme.spec.ts`, where the filter name is updated to match the new label.

imanish003 committed:
Merge branch 'trunk' into fix/html-entities-in-product-name

albarin commented on assets/js/blocks/product-collection/inspector-controls/hand-picked-products-control.tsx:
```diff
@@ -17,6 +17,25 @@ import {
+ * @param {string} str - The string containing HTML entities.
+ * @return {string} - The decoded string.
+ */
+function decodeHTMLEntities( str?: string ) {
```
Hey @imanish003 I think we already have a function that does this, check [here](https://github.com/woocommerce/woocommerce-blocks/blob/4020db4c10d4a3d984e2207285976434bb0b2a89/assets/js/atomic/blocks/product-elements/add-to-cart/product-types/variable/variation-attributes/attribute-select-control.tsx#L78). I think we should probably used that.

imanish003 replied:
Good catch @albarin. Thanks 🙏  Let me quickly change the code.

imanish003 replied:
Made the changes in https://github.com/woocommerce/woocommerce-blocks/pull/11927/commits/70ea5f1a679dd94a685cef39631e406574130d44

imanish003 committed:
Refactor: Replace custom HTML entity decoder with `@wordpress/html-entities`

Rationale:
- The shift to `@wordpress/html-entities` aligns with standard WordPress practices, ensuring consistency across the platform.
- Enhances maintainability by relying on a well-supported library rather than custom code.
- Simplifies the codebase by removing a redundant utility function.

This change enhances the robustness of our code and aligns with best practices in WordPress development.

imanish003 committed:
Merge branch 'fix/html-entities-in-product-name' of https://github.com/woocommerce/woocommerce-blocks into fix/html-entities-in-product-name

albarin reviewed (approved):
Thanks for the update! LGTM 🚀

kmanijak reviewed (approved):
Thanks for a quick fix 🚀 I tested some other entities as well, works as expected!

imanish003 committed:
Merge branch 'trunk' into fix/html-entities-in-product-name
